packages/arb-validator-core/gobridge/challenge.go

```diff
@@ -78,30 +78,31 @@ func (c *challenge) TimeoutChallenge(
 
 func (c *challenge) commitToSegment(hashes [][32]byte) {
 	tree := valprotocol.NewMerkleTree(hashSliceToHashes(hashes))
-	c.challengeData.challengerDataHash = tree.GetRoot()
+	c.challengerDataHash = tree.GetRoot()
 }
 
 func (c *challenge) asserterResponded() {
-	c.challengeData.state = challengerTurn
+	c.state = challengerTurn
 	currentTicks := common.TicksFromBlockNum(c.client.getCurrentBlock().Height)
-	c.challengeData.deadline = currentTicks.Add(c.challengeData.challengePeriodTicks)
+	c.deadline = currentTicks.Add(c.challengePeriodTicks)
 
 }
 
 func (c *challenge) challengerResponded() {
-	c.challengeData.state = asserterTurn
+	c.state = asserterTurn
 	currentTicks := common.TicksFromBlockNum(c.client.getCurrentBlock().Height)
-	c.challengeData.deadline = currentTicks.Add(c.challengeData.challengePeriodTicks)
+	c.deadline = currentTicks.Add(c.challengePeriodTicks)
 
 }
 
 func (c *challenge) resolveChallenge(winner common.Address, loser common.Address) {
-	rollup, ok := c.client.rollups[c.contractAddress]
+	rollup, ok := c.client.rollups[c.rollupAddr]
 	if ok {
-		winningStaker := rollup.rollup.stakers[winner]
+		winningStaker := rollup.stakers[winner]
 		winningStaker.inChallenge = false
-		transferEth(c.client.goEthdata, winner, loser, rollup.rollup.chainParams.StakeRequirement)
-		delete(c.client.rollups[c.contractAddress].rollup.stakers, loser)
+		half := new(big.Int).Div(c.client.ethWallet[winner], rollup.chainParams.StakeRequirement)
```

Comment:
This still isn't right. It should just be stake requirement divided by 2.

Reply:
fixed